src/puppetlabs/dujour/version_check.clj

```diff
@@ -130,7 +133,7 @@
   ([request-values update-server-url callback-fn]
     (validate-config! request-values update-server-url)
     (future
-      (let [key-map (and [:certname :cacert] (keys request-values))
+      (let [key-map (vec (set/intersection (set [:cacert :certname]) (set (keys request-values))))
```

Comment:
It looks like this is just throwing away the other keys in request-values just to get `[:cacert :certname]` each time.